src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs

```diff
@@ -70,13 +70,21 @@ SymbolAccessor symbolAccessor
         bool isStatic
     )
     {
-        var valid =
-            method is { ReturnsVoid: false, IsGenericMethod: false }
-            && (allowPartial || !method.IsPartialDefinition)
-            && (!isStatic || method.IsStatic);
-        return valid && BuildParameters(ctx, method, out var parameters)
-            ? new UserImplementedMethodMapping(method, parameters.Source, parameters.ReferenceHandler)
-            : null;
+        var valid = method is { IsGenericMethod: false } && (allowPartial || !method.IsPartialDefinition) && (!isStatic || method.IsStatic);
```

Comment:
Instead of pattern matching for a single property this code base usually uses direct comparisons `!method. IsGenericMethod`.